systemtest/src/test/java/io/strimzi/systemtest/kafka/ListenersST.java

```diff
@@ -0,0 +1,822 @@
+import io.strimzi.systemtest.utils.kubeUtils.objects.SecretUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+import java.util.Map;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+
+import static io.strimzi.systemtest.Constants.LOADBALANCER_SUPPORTED;
+import static io.strimzi.systemtest.Constants.NODEPORT_SUPPORTED;
+import static io.strimzi.systemtest.Constants.REGRESSION;
+import static io.strimzi.test.k8s.KubeClusterResource.kubeClient;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+@Tag(REGRESSION)
+public class ListenersST extends BaseST {
+    private static final Logger LOGGER = LogManager.getLogger(ListenersST.class);
+
+    public static final String NAMESPACE = "kafka-listeners-cluster-test";
+
+    private String customCertChain1 = "custom-certificate-chain-1";
+    private String customCertChain2 = "custom-certificate-chain-2";
+    private String customCertServer1 = "custom-certificate-server-1";
+    private String customCertServer2 = "custom-certificate-server-2";
+    private String customRootCA1 = "custom-certificate-root-1";
+    private String customRootCA2 = "custom-certificate-root-2";
+
+    @Test
+    @Tag(NODEPORT_SUPPORTED)
+    void testCustomSoloCertificatesForNodePort() throws Exception {
+        String topicName = "test-topic-" + rng.nextInt(Integer.MAX_VALUE);
+
+        KafkaResource.kafkaEphemeral(CLUSTER_NAME, 3, 3)
+            .editSpec()
+                .editKafka()
+                    .editListeners()
+                        .withNewTls()
+                            .withNewConfiguration()
+                                .withNewBrokerCertChainAndKey()
+                                    .withSecretName(customCertServer1)
+                                    .withKey("ca.key")
+                                    .withCertificate("ca.crt")
+                                .endBrokerCertChainAndKey()
+                            .endConfiguration()
+                        .endTls()
+                        .withNewKafkaListenerExternalNodePort()
+                            .withNewConfiguration()
+                                .withNewBrokerCertChainAndKey()
+                                    .withSecretName(customCertServer1)
+                                    .withKey("ca.key")
+                                    .withCertificate("ca.crt")
+                                .endBrokerCertChainAndKey()
+                            .endConfiguration()
+                        .endKafkaListenerExternalNodePort()
+                    .endListeners()
+                .endKafka()
+            .endSpec().done();
+
+        String userName = "alice";
```

Comment:
Move this method variable `userName` to class variable, because you are using mostly in all  test cases as i can see.